packages/webapp-server/src/index.ts

```diff
@@ -1,9 +1,22 @@
 import express from 'express';
+import mongoose from 'mongoose';
 
 const app = express();
 
 app.use(express.json());
 
-app.listen(3001, () => {
-  console.log('Server started on port 3001');
-});
+const start = async () => {
+  try {
+    await mongoose.connect(process.env.MONGODB_URI || '');
+    console.log('Connected to MongoDB');
+  } catch (err) {
+    console.log('Error connecting to MongoDB');
+    console.error(err);
+  }
+
+  app.listen(3001, () => {
+    console.log('Listening on port 3001');
+  });
+};
+
+start();
```

Comment:
Establishing and awaiting the mongoose connection in an async start function ensures that the connection gets established before the app starts listening.

Reply:
why are we not using the `require('dotenv').config();` then destructureing the object for the URI?

Reply:
We do need to do that. I will add that now.

Reply:
One concept we might consider if we have time is creating an "AppConfig" type object that we populate with all .env variables at the start of the app, validate that they all exist, and then use that object to access the variables instead of process.env. It would be injected into any dependencies in the app and could easily be mocked out in testing.